core/names.ts

```diff
@@ -184,15 +184,13 @@ export class Names {
    */
   getDistinctName(name: string, type: NameType|string): string {
     let safeName = this.safeName_(name);
-    let i = '';
-    while (this.dbReverse.has(safeName + i) ||
-           this.reservedWords.has(safeName + i)) {
+    let i: number|null = null;
+    while (this.dbReverse.has(`${safeName}${i ?? ''}`) ||
```

Comment:
Does `safeName + i` or `${saveName + i}` work? It would be nice if there was a way to make this a bit cleaner. The duplication of the template string is a bit difficult to parse.